jerry-core/jrt/jrt.h

```diff
@@ -108,7 +113,11 @@ extern void __noreturn jerry_unreachable (const char *, const char *, const uint
     } \
   } while (0)
 
+#ifdef(__GNUC__)
 #define JERRY_UNREACHABLE() __builtin_unreachable ()
+#else /* !__GNUC__ */
+#define JERRY_UNREACHABLE() abort(); /* but what if they haven't included stdlib?!? */
```

Comment:
I think there's some misunderstanding here :)  
`__builtin_unreachable()` is a built-in function which causes undefined behavior when called during actual execution. Since it's a function that never returns the compiler can make assumptions based on that (but so can the compiler about `abort()` which also never returns).  The current behavior of `JERRY_UNREACHABLE()` works just fine in debug builds with a non-GCC compiler so we don't need to worry about that. However, for release builds we need an alternative to `__builtin_unreachable()` since this is a GCC-specific function.  
`abort()` provides us exactly that: It terminates the program in a controlled manner (rather than just crashing the program) and since `abort()` is part of the C standard library this will work just fine with any C99-compliant compiler.  
Makes sense? :)  
(BTW `assert()` is typically implemented on top of `abort()` e.g. it prints some details about the location of the assertion in the code and then just calls `abort()`)